Dokan/Base/BaseAPI.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  BaseAPI.swift
+//  Dokan
+//
+//  Created by Clara Mounir Adly on 22/06/2022.
+//
+
+import Foundation
+import Alamofire
+
+class BaseAPI<T:TargetType>{
+    private func buildParams(task: Task) -> ([String: Any], ParameterEncoding){
+           switch task {
+           case .requestPlain:
+               return ([:], URLEncoding.default)
+           case .requestParameters(parameters: let parameters, encoding: let encoding):
+               return (parameters, encoding)
+           }
+       }
+
+    func fetchData<M: Decodable>(target: T, responseClass: M.Type, completionHandler:@escaping (Result<M, NSError>)-> Void) {
+          let method = Alamofire.HTTPMethod(rawValue: target.method.rawValue)
+          let headers = Alamofire.HTTPHeaders(target.headers ?? [:])
+          let parameters = buildParams(task: target.task)
+          
+          AF.request(target.baseURL + target.path, method: method, parameters: parameters.0, encoding: parameters.1, headers: headers).responseJSON { (response) in
+              guard let statusCode = response.response?.statusCode else {
+                  print("StatusCode not found")
+                  completionHandler(.failure(NSError()))
+                  return
+              }
+              
+              if statusCode == 200 {
```

Comment:
- Success status codes range from `200 ..< 299` so this may result is false positive.
    - More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
- Alamofire already do this under the hood.